src/audio/component.c

```diff
@@ -159,14 +159,23 @@ int comp_set_state(struct comp_dev *dev, int cmd)
 		}
 		break;
 	case COMP_TRIGGER_STOP:
+		if (dev->state == COMP_STATE_ACTIVE) {
+			dev->state = COMP_STATE_PAUSED;
+		} else {
+			trace_comp_error("comp_set_state() error: "
+					 "wrong state = %u, "
+					 "COMP_TRIGGER_STOP", dev->state);
+			ret = -EINVAL;
+		}
+		break;
 	case COMP_TRIGGER_XRUN:
 		if (dev->state == COMP_STATE_ACTIVE ||
 		    dev->state == COMP_STATE_PAUSED) {
-			dev->state = COMP_STATE_PREPARE;
+			dev->state = COMP_STATE_READY;
```

Comment:
Shouldn't we set COMP_STATE_READY unconditionally? We've detected an xrun, so we would like to reset the whole pipeline.

Reply:
I think that should work, let me change it.